service/src/main/java/bio/terra/profile/service/policy/TpsApiDispatch.java

```diff
@@ -0,0 +1,112 @@
+package bio.terra.profile.service.policy;
+
+import bio.terra.common.logging.RequestIdFilter;
+import bio.terra.policy.api.TpsApi;
+import bio.terra.policy.client.ApiClient;
+import bio.terra.policy.client.ApiException;
+import bio.terra.policy.model.*;
+import bio.terra.profile.app.configuration.PolicyServiceConfiguration;
+import bio.terra.profile.service.policy.exception.*;
+import io.opentelemetry.instrumentation.annotations.WithSpan;
+import java.io.IOException;
+import java.util.UUID;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.MDC;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TpsApiDispatch {
+
+  PolicyServiceConfiguration policyServiceConfiguration;
+
+  @Autowired
+  TpsApiDispatch(PolicyServiceConfiguration policyServiceConfiguration) {
+    this.policyServiceConfiguration = policyServiceConfiguration;
+  }
+
+  private ApiClient getApiClient(String accessToken) {
+    ApiClient client =
+        new ApiClient()
+            .setBasePath(policyServiceConfiguration.getBasePath())
+            .addDefaultHeader(
+                RequestIdFilter.REQUEST_ID_HEADER, MDC.get(RequestIdFilter.REQUEST_ID_MDC_KEY));
+    client.setAccessToken(accessToken);
+    return client;
+  }
+
+  private TpsApi policyApi() {
+    try {
+      return new TpsApi(
+          getApiClient(policyServiceConfiguration.getAccessToken())
+              .setBasePath(policyServiceConfiguration.getBasePath()));
+    } catch (IOException e) {
+      throw new PolicyServiceAuthorizationException(
+          "Error reading or parsing credentials file at %s"
+              .formatted(policyServiceConfiguration.getClientCredentialFilePath()),
+          e.getCause());
+    }
+  }
+
+  // -- Policy Attribute Object Interface --
+  @WithSpan
+  public void createPao(
+      UUID objectId, TpsPolicyInputs inputs, TpsComponent component, TpsObjectType objectType)
+      throws InterruptedException {
+    TpsApi tpsApi = policyApi();
+    try {
+      TpsRetry.retry(
+          () ->
+              tpsApi.createPao(
+                  new TpsPaoCreateRequest()
+                      .objectId(objectId)
+                      .component(component)
+                      .objectType(objectType)
+                      .attributes(inputs)));
+    } catch (ApiException e) {
+      throw convertApiException(e);
+    }
+  }
+
+  @WithSpan
+  public void deletePao(UUID objectId) throws InterruptedException {
+    TpsApi tpsApi = policyApi();
+    try {
+      try {
+        TpsRetry.retry(() -> tpsApi.deletePao(objectId));
+      } catch (ApiException e) {
+        throw convertApiException(e);
+      }
+    } catch (PolicyServiceNotFoundException e) {
+      // Not found is not an error as far as BPM is concerned.
```

Comment:
Finally grokking this, this is if the PAO is not found which makes sense here. The nested try's are a bit awkward.

Reply:
Reworded the comment a bit to hopefully help with that a bit. Agreed that the nested try's are awkward, but I'd rather deal with a converted ApiException (`PolicyServiceNotFoundException`) and not have to duplicate the code in  `convertApiException` that catches 404s